swatch-contracts/src/main/java/com/redhat/swatch/contract/resource/ContractsTestingResource.java

```diff
@@ -321,26 +321,39 @@ public RpcResponse syncAllSubscriptions(Boolean forceSync) throws ProcessingExce
   @RolesAllowed({"test", "support", "service"})
   public OfferingResponse syncOffering(String sku) throws ProcessingException {
     var response = new OfferingResponse();
+    SyncResult result = null;
     try {
       log.info("Sync for offering {}", sku);
-      SyncResult result = offeringSyncService.syncOffering(sku);
+      result = offeringSyncService.syncOffering(sku);
       response.detail(String.format("%s for offeringSku=\"%s\".", result, sku));
     } catch (Exception e) {
-      log.error("Error syncing offering", e.getMessage());
-      if (e.getCause() instanceof ApiException) {
-        ApiException apiException = (ApiException) e.getCause();
+      if (e.getCause() instanceof ApiException apiException) {
```

Comment:
I think this is dead code which is never invoked. What we should do instead is:
- catch the ApiException (thrown by the ProductApi) and in this scenario, return an InternalServerErrorException exception (because the service is handing the connection to the ProductApi
- and don't catch any other exception